### ISSUE TITLE
`ids` on HasManyReference is not autotracked

### ISSUE THREAD

**Issue by YoranBrondsema** [emberjs/data#7324]
### Reproduction
I reproduced the bug in a [Twiddle](https://ember-twiddle.com/1271e3cf6b541c181579027d744c2783?openFiles=templates.application%5C.hbs%2C). When you click on the "Create comment" button, comments are created and they are correctly added to the bullet list. However, the number of comments stays at 0.

### Description
When a property accesses properties from a `HasManyReference`, it is not autotracked. 

```js
get numberOfComments() {
  return this.model.hasMany('comments').ids().length;
}
```

We use this pattern throughout our app to deduce certain properties (is an association empty? how many elements in the association?) without having to sideload the entire collection of associated elements.

Turning it into a computed property makes it work:

```js
@computed('comments.length')
get numberOfComments() {
  return this.model.hasMany('comments').ids().length;
}
```

But ideally we'd like to step away from CP's in our app.

### Versions
The bug is reproduced in the [Twiddle](https://ember-twiddle.com/1271e3cf6b541c181579027d744c2783?openFiles=templates.application%5C.hbs%2C) for Ember.js 3.18.1 and Ember Data 3.18.0.

We are seeing the issue in our app for 3.19:

```cli
└── ember-source@3.19.0 
└── ember-cli@3.19.0 
└── ember-data@3.19.0
```

**Comment by igorT** [comment]
@pzuraq what would be the best way to handle this?

**Comment by runspired** [comment]
We could make some things on references tracked to play nice with these sorts of patterns, however in this case I don't think there's anything we can do short of a large refactor to make reference pull data from a new source that is able to be notified of changes. What we hand back currently is a native array created during the call, so there's nothing there to subscribe to.

We could add tracked to something on the relationship deep in the internals for this case, but with our current architecture and the way we need to interop with the observer world (not to mention our current design direction with the notification manager) that would risk a lot.

**Comment by majnikov** [comment]
Faced with the same.
@runspired any chances to have a fix?

**Comment by runspired** [comment]
closing as duplicate of #7102, we should prioritize a fix for this in 3.28+

**Comment by runspired** [comment]
#7796 will add this feature
